core/src/main/java/com/linecorp/armeria/common/util/UnmodifiableFuture.java

```diff
@@ -113,7 +113,7 @@ public boolean complete(@Nullable T value) {
         throw new UnsupportedOperationException();
     }
 
-    private void doComplete(@Nullable T value) {
+    protected final void doComplete(@Nullable T value) {
```

Comment:
- How about removing `final` so that a subclass can call this method?
- Javadoc